x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverAction.java

```diff
@@ -121,6 +174,27 @@ public RolloverAction(StreamInput in) throws IOException {
         } else {
             maxPrimaryShardDocs = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_8_4_0)) {
+            if (in.readBoolean()) {
+                minSize = new ByteSizeValue(in);
+            } else {
+                minSize = null;
+            }
+            if (in.readBoolean()) {
+                minPrimaryShardSize = new ByteSizeValue(in);
+            } else {
+                minPrimaryShardSize = null;
+            }
```

Comment:
These can be:

```suggestion
            minSize = in.readOptionalWriteable(ByteSizeValue::new);
            minPrimaryShardSize = in.readOptionalWriteable(ByteSizeValue::new);
```

Reply:
Agreed, that's way better, 76249a47a12164b7640b28af05f1c0a3f3f1b910.